odl/discr/lp_discr.py

```diff
@@ -99,6 +99,9 @@ def __init__(self, fspace, partition, dspace, exponent=2.0,
         if not fspace.domain.contains_set(partition.set):
             raise ValueError('Partition {} is not a subset of the function '
                              'domain {}'.format(partition, fspace.domain))
+        if fspace.out_dtype != dspace.dtype:
+            raise ValueError('fspace.out_dtyp {} does not match dspace.dtype '
```

Comment:
I prefer describing the attributes instead of explicitly naming them. Makes it more robust with respect to subclassing.